Tests/SwiftDocCTests/Semantics/MetadataTests.swift

```diff
@@ -433,5 +433,32 @@ class MetadataTests: XCTestCase {
         
         return (problemIDs, metadata)
     }
+
+    func testInvalidMetadataDirectivesInDocumentationCommentHaveSolution() throws {
+        let source = """
+        @Metadata {
+            @DisplayName("Custom Name")
+            @TechnologyRoot
+        }
+        """
+        let document = Document(parsing: source, options: .parseBlockDirectives)
+        let directive = document.child(at: 0)! as! BlockDirective
+        let (bundle, _) = try testBundleAndContext()
+        var problems = [Problem]()
+        let metadata = Metadata(from: directive, source: nil, for: bundle, problems: &problems)
+        
+        metadata?.validateForUseInDocumentationComment(symbolSource: nil, problems: &problems)
+        
+        XCTAssertEqual(problems.count, 2)
+        
+        // Verify that each problem has a solution to remove the directive
+        for problem in problems {
+            XCTAssertNotNil(problem.possibleSolutions)
+            XCTAssertEqual(problem.possibleSolutions?.count, 1)
```

Comment:
`possibleSolutions` is non-optional so the `XCTAssertNotNil` check isn't necessary and the optional chaining in the other assertions fail to compile.

```
XCTAssertEqual(problem.possibleSolutions?.count, 1)
                                        `- error: cannot use optional chaining on non-optional value of type '[Solution]'
```

Reply:
FYI: since you're checking a few different values of the same first solution, you can require that the first solution exist once using `XCTUnwrap` and use the unwrapped value in the later test assertions. 

However, this is just a matter of style and preference. Not a requested change. I'm happy with both styles here, whichever you find more readable.